src/runtime/components/NuxtImg.vue

```diff
@@ -1,14 +1,28 @@
 <template>
   <img
+    v-if="!custom"
     ref="imgEl"
-    :class="props.placeholder && !placeholderLoaded ? props.placeholderClass : undefined"
+    :class="props.placeholder && !isImageLoaded ? props.placeholderClass : undefined"
     v-bind="{
       ...isServer ? { onerror: 'this.setAttribute(\'data-error\', 1)' } : {},
       ...imgAttrs,
       ...attrs,
     }"
     :src="src"
   >
+
+  <slot
+    v-else
+    v-bind="{
+      ...isServer ? { onerror: 'this.setAttribute(\'data-error\', 1)' } : {},
+      imgAttrs: {
+        ...imgAttrs,
+        ...attrs,
+      },
+      isLoaded: isImageLoaded,
```

Comment:
this looks perfect now, just one change - I think let's not expose `isImageLoaded` as this can be misleading - on server load the load may take place before hydration and this will never end up being set to `true`

Reply:
if it's significant, we could also open a new PR to implement the functionality - maybe through an approach similar to error handling?